src/iter.rs

```diff
@@ -15,7 +15,7 @@ pub type HexSliceToBytesIter<'a> = HexToBytesIter<HexDigitsIter<'a>>;
 
 /// Iterator yielding bytes decoded from an iterator of pairs of hex digits.
 pub struct HexToBytesIter<T: Iterator<Item = [u8; 2]>> {
-    iter: T,
+    iter: Enumerate<T>,
```

Comment:
I think it'd be better to store the original length of the string and then compute position as `original_len - iter.len() * 2`. The reason is we probably want a fast success path and having to increase a number every loop iteration in addition to increasing the pointer inside iterator is waste of performance.

This needs `ExactSizeIterator` bound but that should be fine.